301-metadata-repositories.md

```diff
@@ -0,0 +1,158 @@
+---
+title: CNAB Security: Metadata repositories
+weight: 301
+---
+
+# Cloud Native Application Bundles Security (CNAB-Sec): Metadata repositories 1.0 WD
+
+* [Metadata repositories](#metadata-repositories)
+* [Minimum viable product (MVP)](#minimum-viable-product-mvp)
+  * [Security analysis of MVP](#security-analysis-of-mvp)
+
+This document is a _normative_ part of [CNAB Security](300-CNAB-security.md).
+
+The keywords MUST, MUST NOT, REQUIRED, SHALL, SHALL NOT, SHOULD, SHOULD NOT, RECOMMENDED, MAY, and OPTIONAL in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119). The use of these keywords in this document are statements about how a CNAB implementation may fulfill the CNAB Security specification _only_.
+
+## Metadata repositories
+
+A _metadata repository_ is a service that hosts The Update Framework (TUF) and in-toto metadata about bundles. As discussed in [300](300-CNAB-security.md), TUF adds signed metadata that provides authenticity and integrity of bundles between [CNAB registries](200-CNAB-registries.md) and users, whereas in-toto adds signed metadata that provides authenticity and integrity of bundles between developers and CNAB registries. By combining both types of signed metadata, we get end-to-end authenticity and integrity of bundles between developers and users.
+
+Note that:
+* A metadata repository MAY be physically distinct from a CNAB registry, or not (e.g., bundles as well as TUF and/or in-toto metadata MAY live as [OCI Artifacts](https://stevelasker.blog/2019/08/25/oci-artifacts-and-a-view-of-the-future/) on a CNAB registry).
+* Both TUF and in-toto are frameworks that can be configured in a wide variety of ways to achieve varying degrees of security. However, we discuss one [minimum viable product (MVP)](#minimum-viable-product-mvp), which has been optimized in particular for [Docker Content Trust / Notary 0.6.1](https://github.com/theupdateframework/notary/releases/tag/v0.6.1), in this document.
+* Although we discuss only the signing and verification of bundles, exactly the same principles apply to images.
+
+## Minimum viable product (MVP)
+
+This subsection discusses the simplest way that developers MAY set up a metadata repository for their bundle. Every bundle MAY use a separate metadata repository on the same server, even if two or more bundles are maintained by the same group of developers. (This is similar to how images are signed using [Docker Content Trust](https://docs.docker.com/engine/security/trust/content_trust/). As noted earlier, the MVP has been optimized for [Docker Content Trust / Notary 0.6.1](https://github.com/theupdateframework/notary/releases/tag/v0.6.1).) Figure 1 illustrates our simple metadata repository for a bundle.
```

Comment:
> Every bundle MAY use a separate metadata repository on the same server, [...]

I'd like this document to describe the implications of using _multiple_ servers to host metadata repositories. A usecase is where a user consumes bundles from multiple vendors who do not share a single, central server.

Reply:
I can add a note here about how developers may setup different metadata repos on different servers. And then, in 303 (verification workflows), I'll talk about how a bundle runtime can pull different bundles from different metadata repos on different servers. Would this help?

Reply:
I think so, especially if there are some non-obvious gotchas.

Reply:
Does the new note help, or would you like more details?

Reply:
It helps a little, but don't think the document really describes the security implications of using multiple servers to host metadata repositories. For example, it would be good to spell out how to establish trust in a new metadata repository - presumably mainly by reputation and guess-work? You see, I think this is a weakness of this whole approach in that scenario and it would be good to call that out so that there are no surprises down the line.